src/spikeinterface/comparison/groundtruthstudy.py

```diff
@@ -1,327 +1,405 @@
 from pathlib import Path
 import shutil
+import os
+import json
+import pickle
 
 import numpy as np
 
-from spikeinterface.core import load_extractor
-from spikeinterface.extractors import NpzSortingExtractor
-from spikeinterface.sorters import sorter_dict, run_sorters
+from spikeinterface.core import load_extractor, extract_waveforms, load_waveforms
+from spikeinterface.core.core_tools import SIJsonEncoder
+
+from spikeinterface.sorters import run_sorter_jobs, read_sorter_folder
 
 from spikeinterface import WaveformExtractor
 from spikeinterface.qualitymetrics import compute_quality_metrics
 
-from .paircomparisons import compare_sorter_to_ground_truth
+from .paircomparisons import compare_sorter_to_ground_truth, GroundTruthComparison
 
-from .studytools import (
-    setup_comparison_study,
-    get_rec_names,
-    get_recordings,
-    iter_working_folder,
-    iter_computed_names,
-    iter_computed_sorting,
-    collect_run_times,
-)
 
+# TODO later : save comparison in folders when comparison object will be able to serialize
 
-class GroundTruthStudy:
-    def __init__(self, study_folder=None):
-        import pandas as pd
 
-        self.study_folder = Path(study_folder)
-        self._is_scanned = False
-        self.computed_names = None
-        self.rec_names = None
-        self.sorter_names = None
+# This is to separate names when the key are tuples when saving folders
+_key_separator = " ## "
+# This would be more funny
+# _key_separator = " (°_°) "
+
+
+class GroundTruthStudy:
+    """
+    This class is an helper function to run any comparison on several "cases" for several ground truth dataset.
+
+    "cases" can be:
+      * several sorter for comparisons
+      * same sorter with differents parameters
+      * parameters of comparisons
```

Comment:
What do you mean by this? Maybe we could list:

* different preprocessing methods